### ISSUE TITLE
Fusion: MapImplementation could check type of `$collection` before `count()` to avoid PHP Fatal error

### ISSUE THREAD

**Issue by sorenmalling** [neos/neos-development-collection#3359]
If the `items` of a Neos.Fusion:Loop is a empty string (not Countable), the `count(..)` call returns a PHP Fatal Error
```
Warning: count(): Parameter must be an array or an object that implements Countable in /Users/soren/Projects/findwebshop.dk/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Neos_Fusion_FusionObjects_MapImplementation.php line 87
```

Should/could we throw a InvalidArgumentException, return blank, return empty array - all to avoid the PHP fatal error

**Comment by mhsdesign** [comment]
Same for:
assume i have this totally not copied object from somewhere:

```php
class DatesCollection implements IteratorAggregate {
  private $dates;

  public function __construct(\DateTimeImmutable ...$dates) {
    $this->dates = $dates;
  }

  public function getIterator() {
     return new ArrayIterator($this->dates);
  }
}
```
this collection *is* perfectly loopable, but still fusion fails, since it tries to `count` it first (to have more information, whether this is the last item etc -> which should be available already via the iterator idk?):
https://github.com/neos/neos-development-collection/blob/18e16b84186edc7ab9793ec1a3345bfc123593f7/Neos.Fusion/Classes/FusionObjects/MapImplementation.php#L87

**Comment by mhsdesign** [comment]
fixed via: #4168
